### ISSUE TITLE
MarkerTapListener and Callout

### ISSUE THREAD

**Issue by dferg7** [moagrius/TileView#433]
In order to implement a somewhat onLongClick functionality I was previously told to extend the TileView and override onLongPress. Having done so, I'm now wondering how I can add in a Marker Tap Listener to display information about the markers that are clicked on by the user. Do I have to make my own Callout activity in order to show callouts on the screen? Or is there any other ways of doing this?

**Comment by moagrius** [comment]
not sure what you're question is exactly.  the demo app shows how to use markers and callouts, and the readme has a section here: https://github.com/moagrius/TileView#markers--callouts

there's also the wiki and the api docs.  what specifically are you asking?

**Comment by dferg7** [comment]
Sorry, let me rephrase this. I looked into it a little more and I now have a MarkerTapListener for when the user clicks on the marker that is placed on the tile. I guess my question is, if I'm placing markers by overriding the onLongPress functionality, how would I later acces the information for that marker when the user clicks on it? 

In TileViewExtended is where I override onLongPress and place a marker:
```

@Override
    public void onLongPress(MotionEvent event) {
        float scale = getScale();
        relX = getCoordinateTranslater().translateAndScaleAbsoluteToRelativeX(getScrollX() + event.getX(), scale);
        relY = getCoordinateTranslater().translateAndScaleAbsoluteToRelativeY(getScrollY() + event.getY(), scale);

         ImageView marker = new ImageView(MainActivity.getAppContext());
         marker.setImageResource(R.drawable.black_location);
         addMarker(marker, relX, relY, -0.5f, -1.0f);
}
```

In my main activity: I set up a marker tap listener  (note I didn't include some code because this is the basic setup for the tile view)

        ....
        tileView = new TileViewExtended(this);
        tileView = (TileViewExtended) findViewById(R.id.tileview);
        ....
        tileView.getMarkerLayout().setMarkerTapListener(markerTapListener);
```

public TileViewExtended getTileView() {
           return tileView
}
```

```
    private MarkerLayout.MarkerTapListener markerTapListener = new MarkerLayout.MarkerTapListener() {
        @Override
        public void onMarkerTap(View view, int x, int y) {
            TileViewExtended tileView = getTileView();
            Toast.makeText( getApplicationContext(), "You tapped a marker", Toast.LENGTH_LONG ).show();
        }
 };
```

Right now I only having it showing a toast for when a marker is clicked on.

**Comment by moagrius** [comment]
this is not specific to TileView.  just like you would with a standard View and View.OnClickListener, there are many ways to get what you want.  you could associate the `View` with an identifier or listener in a `Map`, or more commonly use `setTag`to identify it somehow on the `View`, and check that in the listener.  this is all done in the the demo - each marker is given some identifier with `setTag`, and in the listener that tag is read with `getTag`.  take a look at the demo source
